src/main/java/dk/cngroup/trainings/spring/springassignment/service/AnimalServiceImpl.java

```diff
@@ -32,9 +32,9 @@ public AnimalServiceImpl(AnimalRepository animalRepository, CareTakerRepository
 	}
 
 	@Override
-	public Optional<Animal> getAnimalById(long id) throws AnimalNotFoundException {
+	public Animal getAnimalById(long id) throws AnimalNotFoundException {
 		if (animalRepository.existsById(id)) {
-			return animalRepository.findById(id);
+			return animalRepository.findById(id).get();
```

Comment:
well again, here you are violating the Optional contact, that you will not call get() before isPresent :D... just a little refactoring here should solve it